WordPress/wporg-mu-plugins
# WIP: Block to pull news posts

See https://github.com/WordPress/wporg-main-2022/issues/31.

This PR adds a block: `wporg/multisite-latest-posts`. 

The block makes a server-side (dynamic block) request via `wp_remote_get` to retrieve posts from a WordPress REST API. 

Since we plan to use this within our network, it was suggested by @dd32 that the block first try switching blogs. This is definitely a good idea but I think we can consider that behavior a next iteration (Feel free to disagree!).

This block is also fetching categories in a separate HTTP request because the news blog's API is not embedding the `podcast` categories properly. If fixes are merged, we should update this code to use `?_embed=true`.  

# Note-Worthy
- This block tries to inherit CSS variables that exist in the wporg-parent-2021 [ theme.json](https://github.com/WordPress/wporg-parent-2021/blob/trunk/source/wp-content/themes/wporg-parent-2021/theme.json).
- Doesn't expose styling controls on `html` elements. Ie: Can't change font size via editor.
- The current URL defaults to the news API endpoint :)
- The designs have the post titles and the content below by `15px` but we don't have that preset so I used `10px`.

# Default Frontend View 
<img width="636" alt="Screen Shot 2022-08-02 at 3 34 51 PM" src="https://user-images.githubusercontent.com/1657336/182307972-28eb6a58-a05a-499f-b302-cdea428d5626.png">


# Features
- User can input a WordPress REST API for requests.
- User can change the number of items displayed.

## Editor
<img width="937" alt="Screen Shot 2022-08-02 at 12 38 33 PM" src="https://user-images.githubusercontent.com/1657336/182287649-4e649300-62ed-4adc-a20e-a1b1117a39b7.png">

## Parent Theme Properties
We will need to add these to the `custom` section of the theme.json for the parent theme.

```
"multisite-latests-post": {
	"link": {
		"color": "var(--wp--preset--color--charcoal-1)",
		"spacing": "var(--wp--preset--spacing--10)",
		"details": {
			"fontSize": "var(--wp--preset--font-size--small)"
		}
	},
	"spacing":  "var(--wp--preset--spacing--40)",
	"title": {
		"fontFamily": "var(--wp--preset--font-family--eb-garamond)",
		"fontSize": "var(--wp--preset--font-size--heading-5)",
		"lineHeight": "var(--wp--custom--heading--level-3--typography--line-height)"
	}
}
```


# Testing

1. Add this block to your environment
2. Add block into content
3. Expect to see News posts
4. Change URL to `wordpress.org/showcase/wp-json/wp/v2`
5. Change "Items to Show", expect to see component update in editor (and on page)
